src/macrofier/engine.mjs

```diff
@@ -324,8 +324,7 @@ const generateMacros = (obj, templates, languages, options = {}) => {
 
   const imports = generateImports(obj, templates)
   const initialization = generateInitialization(obj, templates)
-  const enums = generateEnums(obj, templates)
-  const enumsConversion = generateEnumsConversion(obj, templates)
+  const enums = (options.destination && options.destination.endsWith(".cpp") ? generateEnumsConversion(obj, templates) : generateEnums(obj, templates))
```

Comment:
updated to pick based on suffix.

Reply:
But the template replacement part is inside convertEnumTemplate, since it is handling enums from array, properties etc too. But I have taken care the main points you have suggested here